shell/config/router/navigation-guards/clusters.js

```diff
@@ -1,61 +1,21 @@
 import { DEFAULT_WORKSPACE } from '@shell/config/types';
-import { applyProducts } from '@shell/store/type-map';
 import { ClusterNotFoundError, RedirectToError } from '@shell/utils/error';
 import { get } from '@shell/utils/object';
 import { AFTER_LOGIN_ROUTE, WORKSPACE } from '@shell/store/prefs';
-import { BACK_TO } from '@shell/config/local-storage';
 import { NAME as FLEET_NAME } from '@shell/config/product/fleet.js';
 import { validateResource, setProduct } from '@shell/utils/auth';
-import { getClusterFromRoute, getProductFromRoute, getPackageFromRoute } from '@shell/utils/router';
+import { getClusterFromRoute, getProductFromRoute, getPackageFromRoute, routeRequiresAuthentication } from '@shell/utils/router';
 
-let beforeEachSetup = false;
-
-export default async function({
-  route, store, redirect, from, $plugin, next
-}) {
-  if ( store.getters['auth/enabled'] !== false && !store.getters['auth/loggedIn'] ) {
-    return;
-  }
-
-  const backTo = window.localStorage.getItem(BACK_TO);
-
-  if (backTo) {
-    window.localStorage.removeItem(BACK_TO);
+export function install(router, context) {
+  router.beforeEach((to, from, next) => loadClusters(to, from, next, context));
+}
 
-    window.location.href = backTo;
+export async function loadClusters(to, from, next, { store }) {
+  if (!routeRequiresAuthentication(to)) {
+    return next();
   }
 
-  // GC should be notified of route change before any find/get request is made that might be used for that page
-  store.dispatch('gcRouteChanged', route);
-
-  // Load stuff
-  let localCheckResource = false;
-
-  await applyProducts(store, $plugin);
-
-  // Setup a beforeEach hook once to keep track of the current product
-  if ( !beforeEachSetup ) {
-    beforeEachSetup = true;
-    // This only needs to happen when beforeEach hook hasn't run (the initial load)
-    localCheckResource = true;
-
-    store.app.router.beforeEach((to, from, next) => {
-      // NOTE - This beforeEach runs AFTER this middleware. So anything in this middleware that requires it must set it manually
-      setProduct(store, to);
-
-      next();
-    });
-
-    // Call it for the initial pageload
-    setProduct(store, route);
-
-    store.app.router.afterEach((to, from) => {
-      // Clear state used to record if back button was used for navigation
-      setTimeout(() => {
-        window._popStateDetected = false;
-      }, 1);
-    });
-  }
+  const route = to;
```

Comment:
Why this assignment? Are we trying to clone or what? Also, the name in such cases should have more specificity and a comment with an explanation.

Reply:
I only wanted to make this easier to diff with the original implementation so it was obvious the logic wasn't being changed. I change all references.

Reply:
Updated.